### PR TITLE
[Wiki] Update telemetry article

### PR DESCRIPTION
**Pull request by matebarabas**
# Description

Updated telemetry description for the following wiki article: https://github.com/Azure/ResourceModules/wiki/The%20library%20-%20Module%20design#telemetry

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: docs/wiki/The library - Module design.md [hunk 1]
```diff
@@ -671,6 +689,8 @@ When consuming the modules outside of CARML's pipelines you can either
 - Set the parameter to a default value of `'false'`
 - Set the parameter to false when deploying a module
 
-> **Note:** _The deployment and its GUID can NOT be used to track [Azure Consumed Revenue (ACR)](https://docs.microsoft.com/en-us/azure/marketplace/azure-partner-customer-usage-attribution)._
->
-> _If you want to track consumption, we recommend to implement it on the consuming template's level (i.e., the multi-module solution, such as workload/application) and apply the required naming format `'pid-<GUID>'` (without the suffix)._
+## Telemetry vs Customer Usage Attribution
+
+Though similar in principles, this approach is not to be confused and does not conflict with the usage of CUA IDs that are used to track Customer Usage Attribution of Azure marketplace solutions (partner solutions). The GUID-based telemetry approach described here can coexist and can be used side-by-side with CUA IDs. If you have any partner or customer scenario require the addition of CUA IDs, you can customize the modules of this library by adding the required CUA ID deployment while keeping the built-in telemetry solution.
```
Comment (ahmadabdalla) [inline docs/wiki/The library - Module design.md:694]: "If you have any partner or customer scenarios that require the addition of CUA IDs"

File: docs/wiki/The library - Module design.md [hunk 2]
```diff
@@ -655,13 +659,27 @@ In addition, they follow these file-type-specific guidelines:
 
 # Telemetry
 
+## Overview 
+
+Microsoft uses the approach detailed in this section to identify the deployments of the Bicep and ARM JSON templates of the CARML library. Microsoft collects this information to provide the best experiences with their products and to operate their business. Telemetry data is captured through the built-in mechanisms of the Azure platform; therefore, it never leaves the platform, providing only Microsoft with access. Deployments are identified through a specific GUID (Globally Unique ID) representing the code is originated in the CARML library. The data is collected and governed by Microsoft's privacy policies, located at the [Trust Center](https://www.microsoft.com/en-us/trust-center).
```
Comment (ahmadabdalla) [inline docs/wiki/The library - Module design.md:664]: "Deployments are identified through a specific GUID (Globally Unique ID), indicating that the code originated from the CARML library"

File: docs/wiki/The library - Module design.md [hunk 3]
```diff
@@ -655,13 +659,27 @@ In addition, they follow these file-type-specific guidelines:
...
 Each module in CARML contains a `defaultTelemetry` deployment `'pid-<GUID>-${uniqueString(deployment().name)}'`, resulting in deployments such as `'pid-<GUID>-nx2c3rnlt2wru'`.
 
 > **Note:** Though implemented at each level in a module hierarchy (e.g., storage account & blobServices), the deployment will only happen for the top-level module in a deployment, but not for its children. To illustrate this better, see the following examples:
 >
 > - Deployment of the KeyVault module and 2 Secrets: Results in 1 `PID` deployment for the KeyVault (and none for the secrets)
 > - Deployment of the Secret module: Results in 1 `PID` deployment for the Secret
 
+## Opting Out
+
+Albeit telemetry described in this section is optional, the implementation follows an opt-out logic, as most commercial software solutions, this project also requires continuously demonstrating evidence of usage, hence the CARML product team recommends leaving the telemetry setting on its default, enabled configuration.
```
Comment (ahmadabdalla) [inline docs/wiki/The library - Module design.md:681]: Maybe we should rephrase the "CARML product team" bit?

File: docs/wiki/The library - Module design.md (hunk 1 above)
Comment (matebarabas) [reply docs/wiki/The library - Module design.md]: Updated as suggested.

File: docs/wiki/The library - Module design.md (hunk 2 above)
Comment (matebarabas) [reply docs/wiki/The library - Module design.md]: Updated as suggested.

File: docs/wiki/The library - Module design.md (hunk 3 above)
Comment (matebarabas) [reply docs/wiki/The library - Module design.md:681]: This is intentionally phrased as is.

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**